components/measure-upload/MeasureRepositoryUpload.tsx

```diff
@@ -0,0 +1,235 @@
+import { Button, Collapse, Grid, Group, Loader, Select, TextInput } from '@mantine/core';
+import React, { useState } from 'react';
+import { useEffect } from 'react';
+import { useRecoilState, useRecoilValue, useSetRecoilState } from 'recoil';
+import { measureBundleState } from '../../state/atoms/measureBundle';
+import { measurementPeriodState } from '../../state/atoms/measurementPeriod';
+import { identifyMissingValueSets, MeasureUploadProps, populateMeasurementPeriod } from '../../util/measureUploadUtils';
+import { showNotification } from '@mantine/notifications';
+import { v4 as uuidv4 } from 'uuid';
+import { IconAlertCircle } from '@tabler/icons';
+import { CircleCheck } from 'tabler-icons-react';
+import { displayMapToSelectDataState } from '../../state/selectors/displayMapToSelectData';
+
+enum PACKAGE_STATES {
+  LOADING,
+  SUCCESS,
+  FAIL,
+  NONE
+}
+
+const MEASURE_RETRIEVAL_BODY = {
+  resourceType: 'Parameters',
+  parameter: [
+    {
+      name: 'include-terminology',
+      valueBoolean: true
+    }
+  ]
+};
+
+const MEASURE_RETRIEVAL_OPTIONS = {
+  method: 'POST',
+  body: JSON.stringify(MEASURE_RETRIEVAL_BODY),
+  headers: { 'Content-Type': 'application/json+fhir' }
+};
+
+export default function MeasureRepositoryUpload({ logError }: MeasureUploadProps) {
+  const [isLoadingIds, setIsLoadingIds] = useState(false);
+  const [isLoadingPackage, setIsLoadingPackage] = useState<PACKAGE_STATES>(PACKAGE_STATES.NONE);
+  const [measureSelectIsOpen, setMeasureSelectIsOpen] = useState(false);
+
+  const [{ measureRepositoryUrl, isFile, selectedMeasureId, displayMap }, setMeasureBundle] =
+    useRecoilState(measureBundleState);
+  const setMeasurementPeriod = useSetRecoilState(measurementPeriodState);
+
+  const idSelectData = useRecoilValue(displayMapToSelectDataState);
+
+  useEffect(() => {
+    if (isFile) {
+      setMeasureSelectIsOpen(false);
+      setIsLoadingPackage(PACKAGE_STATES.NONE);
+    }
+  }, [isFile]);
+
+  const rejectUpload = (
+    message: string | string[],
+    attemptedBundleDisplay: string | null,
+    isValueSetMissingError = false
+  ) => {
+    logError({
+      id: uuidv4(),
+      message,
+      timestamp: new Date().toISOString(),
+      attemptedBundleDisplay,
+      isValueSetMissingError
+    });
+    showNotification({
+      icon: <IconAlertCircle />,
+      title: 'Bundle upload failed',
+      message: 'There was an issue uploading your bundle. Please correct the issues listed below.',
+      color: 'red'
+    });
+    setIsLoadingPackage(PACKAGE_STATES.FAIL);
+  };
+
+  /*
+    Handles updating of the Measure Repository Service url field and associated state variables
+  */
```

Comment:
```suggestion
  /**
   * Handles updating of the Measure Repository Service url field and associated state variables
  */
```
Not sure if this lines up but I think this is what we have in the rest of this repo